### PR TITLE
feat: add LightGBMRanker

### PR DESCRIPTION
**Pull request by JoanFM**
This change motivates partially this other PR:

https://github.com/jina-ai/jina/pull/1718

**TODO in core**
- Allow batching for Rankers (Depending on https://github.com/jina-ai/jina/pull/2145)

### CODE REVIEW

File: rankers/LightGBMRanker/__init__.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+        The name of the features are used to extract the features from incoming `documents`. Check how these features are accessed in
+        :class:`Document`
+
+    """
+
+    def __init__(self,
+                 model_path: Optional[str] = 'tmp/model.txt',
+                 query_feature_names: Tuple[str] = ['tags__query_length', 'tags__query_language'],
+                 match_feature_names: Tuple[str] = ['tags__document_length', 'tags__document_language',
+                                                    'tags__document_pagerank'],
```
Comment (nan-wang) [inline rankers/LightGBMRanker/__init__.py:39]: I'm not sure whether the default values are proper. At least, we need to mention the default values in the docstring.
```suggestion
                 query_feature_names: Tuple[str] = ['tags__query_length', 'tags__query_language'],
                 match_feature_names: Tuple[str] = ['tags__document_length', 'tags__document_language',
                                                    'tags__document_pagerank'],
```

File: rankers/LightGBMRanker/__init__.py [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+    from model_path
+    :param query_categorical_features: name of features contained in `query_feature_names` corresponding to categorical features.
+    :param match_categorical_features: name of features contained in `match_feature_names` corresponding to categorical features.
+    :param query_features_before: True if `query_feature_names` must be placed before the `match` ones in the `dataset` used for prediction.
+    :param args: Additional positional arguments
+    :param kwargs: Additional keyword arguments
+
+    .. note::
+        The name of the features are used to extract the features from incoming `documents`. Check how these features are accessed in
+        :class:`Document`
```
Comment (nan-wang) [inline rankers/LightGBMRanker/__init__.py:31]:
```suggestion
        :class:`Document` at https://docs.jina.ai/api/jina.types.document/
```

**Review by nan-wang — CHANGES_REQUESTED** [review]
minor changes

File: rankers/LightGBMRanker/__init__.py (hunk 1 above)
Comment (JoanFM) [reply rankers/LightGBMRanker/__init__.py:39]: there are no really good default values

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Comment by nan-wang** [comment]
https://github.com/jina-ai/jina-hub/pull/3670/checks?check_run_id=2111078713#step:6:689

The model file is missing

**Comment by JoanFM** [comment]
> https://github.com/jina-ai/jina-hub/pull/3670/checks?check_run_id=2111078713#step:6:689
> 
> The model file is missing

Check now

**Review by nan-wang — APPROVED** [review]
LGTM👍

**Pull request merged**